dlt-database/entity/0001-init_db/Account.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  Entity,
+  PrimaryGeneratedColumn,
+  Column,
+  CreateDateColumn,
+  ManyToOne,
+  JoinColumn,
+  OneToOne,
+  OneToMany,
+  ManyToMany,
+  JoinTable,
+} from 'typeorm'
+import { User } from './User'
+import { Community } from './Community'
+import { TransactionDraft } from './TransactionDraft'
+import { ConfirmedTransaction } from './ConfirmedTransaction'
+
+@Entity('accounts')
+export class Account {
+  @PrimaryGeneratedColumn('increment', { unsigned: true })
+  id: number
+
+  @ManyToOne(() => User, (user) => user.accounts) // Assuming you have a User entity with 'accounts' relation
+  @JoinColumn({ name: 'user_id' })
+  user: User
+
+  // if user id is null, account belongs to community gmw or auf
+  @Column({ name: 'user_id', type: 'int', unsigned: true, nullable: true })
+  userId?: number
```

Comment:
A user can have a community controlled account, for contributions, only one per user allowed.
A user can have many community controlled accounts for example for business, or traveling but no contribution allowed.
A user can have many crypto accounts, but no contribution allowed.
Access to other accounts I wouldn't model here. By Crypto Accounts the user simply get the keys and by community accounts this will be entirely controlled from backend. So from point of view of blockchain, no account sharing happen at all.

Reply:
So what's the reason for the extra relation users.account_id?
I cant get the first account, the contribution account with 
```sql
select * from accounts where user_id = ? AND derivation_index = 1;
``` 
or
```sql
select * from accounts where user_id = ? AND type = 2;
```
type 2 = contribution

Reply:
Sorry, i think you're designing the data-navigations too much in relational model than in object model.

Imagine the following object model:
  User --- 1:1-Relation AGE-Account ---> Account

with the following relational model:
  users.ageAccountId --> accounts.id

if you search for a User by userId and add in the TypeORM query the expression relations: ['AGE-Account'], you will get an User-Entity with direct access on the one and only preloaded age-account. you have nothing else to filter or select by type from a returnd array of accounts.

For all other account types of a user without contributions you're right to search with a select * from account where account.userId = ?

What do you think about such a modelling?

Reply:
If it is used often, it makes sense.
But I don't see where I would use this.
I have try to make a overview about dlt connector business logic. 
Do you see a place where using a extra relation for age account is useful? 
![DLT business logic overview](https://github.com/gradido/gradido/blob/iota5_dlt-database/docu/Concepts/DLT/overview.md]

Reply:
yes i do. 
on each "create transaction" invokation from backend you have the step "load or create account" PER USER. And this criteria "per user" is exactly what you need twice in each "create transaction" processing. Search and load a user with its related age-account and this happens for the transaction sender and transaction recipient, right?